pymatgen/entries/compatibility.py

```diff
@@ -407,9 +411,127 @@ def __str__(self):
         return "{} {} Correction".format(self.name, self.compat_type)
 
 
-class Compatibility(MSONable):
+class Compatibility(MSONable, metaclass=abc.ABCMeta):
     """
-    The Compatibility class combines a list of corrections to be applied to
+    Abstract Compatibility class, not intended for direct use.
+    Compatibility classes are used to correct the energies of an entry or a set
+    of entries. All Compatibility classes must implement .get_adjustments method.
+    """
+    @abc.abstractmethod
+    def get_adjustments(self, entry: ComputedEntry):
+        """
+        Get the energy adjustments for a ComputedEntry.
+
+        This method must generate a list of EnergyAdjustment objects
+        of the appropriate type (constant, composition-based, or temperature-based)
+        to be applied to the ComputedEntry, and must raise a CompatibilityError
+        if the entry is not compatible.
+
+        Args:
+            entry: A ComputedEntry object.
+
+        Returns:
+            [EnergyAdjustment]: A list of EnergyAdjustment to be applied to the
+                Entry.
+
+        Raises:
+            CompatibilityError if the entry is not compatible
+        """
+        return
+
+    def process_entries(self, entries: Union[ComputedEntry, list], clean: bool = False):
+        """
+        Process a sequence of entries with the chosen Compatibility scheme.
+
+        Args:
+            entries: ComputedEntry or [ComputedEntry]
+            clean: bool, whether to remove any previously-applied energy adjustments.
+                If True, all EnergyAdjustment are removed prior to processing the Entry.
+                Default is False.
+
+        Returns:
+            A list of adjusted entries.  Entries in the original list which
+            are not compatible are excluded.
+        """
+        # convert input arg to a list if not already
+        if isinstance(entries, ComputedEntry):
+            entries = [entries]
+
+        processed_entry_list = []
+
+        if clean:
+            for e in entries:
+                # remove any previously documented corrections from the entry
+                e.energy_adjustments = []
+
+        # apply the corrections
+        for entry in entries:
+            try:
```

Comment:
This try should except should just be around getting the adjustments from the child implementation. Everything else after should be outside the try/catch.

Reply:
Per the abstract class, `process_entries` is supposed to catch all `CompatibilityError`. Since we want to raise a `CompatibilityError` when we find a duplicate correction, are you saying that duplicate checking should be in a separate try/except block?

In other words, we might get `CompatibilityError` from the child method because an entry is incompatible, or we might get a `CompatibilityError` within `process_entries` from duplicate checking.

Reply:
If you don't want process_entry to raise an error, then don't. Purposefully throwing an error to just catch it and then remove it's `Exception` quality is awkward code. Use the logger and print out an error message instead.

Reply:
That makes sense, but I'm having trouble finding an alternative. The reason for raising and catching the exception within `process_entries` was to ensure that, if we found a collision between two energy adjustments, that `Entry` would not be returned. One of the key features of our interface here is that  "if anything goes wrong when correcting an Entry, you get a `CompatibilityError`. That's why I originally put all of that logic into a single try/except block.